ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/TableSelectBuilder.java

```diff
@@ -121,7 +119,8 @@ private TableSelectBuilder() {
 
     final KTable<K, GenericRow> transFormedTable = table.getTable().transformValues(
         () -> new KsTransformer<>(selectMapper.getTransformer(logger)),
-            selectName
+        Materialized.with(keySerde, valSerde),
```

Comment:
Same question as above regarding whether this is adding a new state store, and the implications for backwards incompatibility. 

Also, what's the purpose of moving the building of `physicalSchema`, `keySerde` and `valueSerde` out of the if-statement if they're only used inside the if-statement anyway?

Reply:
Ok. Actually it does not work to move it into the join-builder and we want to set the serdes here, as a `TableSelect` will produce the input-tables for the join (at least in the examples I did -- I assume we can relay on this -- would be good to get confirmation).

Reply:
> out of the if-statement if they're only used inside the if-statement anyway?

Maybe it was not clear from the previous diff, but we did (and still do) use them outside :)

Reply:
Still looking into this in more detail but I don't understand why we can't pass in serdes in the join builder. I also don't understand why passing in serdes in TableSelectBuilder is sufficient for ensuring serdes are always available at the FK join step. I think you might see a serde failure if you uncomment one of the tests in my PR for adding an FK join at the start of an n-way join. Can we sync offline to make this discussion go faster?

Reply:
It was rather a limitation of QQT that tracks registered serdes -- it assigns a unique ID for each registered serde and assumes that each operator registers at most one (output) serde. When we try to register left-value-input and left-value-ouptut serde, it raises an error that we try to register two output serdes.